torchao/prototype/dtypes/uintgen.py

```diff
@@ -0,0 +1,362 @@
+import torch
```

Comment:
Did you have a look at @vayuda's bitpacking code in `prototype/common` -this seems very similar